pkg/resource/reconciler.go

```diff
@@ -320,8 +326,8 @@ func (r *ClaimReconciler) Reconcile(req reconcile.Request) (reconcile.Result, er
 		return reconcile.Result{Requeue: false}, errors.Wrap(IgnoreNotFound(r.client.Status().Update(ctx, claim)), errUpdateClaimStatus)
 	}
 
+	class := r.newClass()
```

Comment:
Should we keep `class` scoped to the `if !meta.WasCreated(managed) {}` block, given it's not used elsewhere?

Reply:
Definitely, good catch!